File: app/src/main/java/cn/intret/app/picgo/app/CoreModule.java
package cn.intret.app.picgo.app;

import android.content.Context;

public class CoreModule {
    private static final CoreModule ourInstance = new CoreModule();

    public static CoreModule getInstance() {
        return ourInstance;
    }

    private CoreModule() {
    }

    private Context mAppContext;
    void init(Context appContext) {
        mAppContext = appContext;
    }

    public Context getAppContext() {
        return mAppContext;
    }
}

File: app/src/main/java/cn/intret/app/picgo/app/CoreModule.kt
package cn.intret.app.picgo.app

import android.annotation.SuppressLint
import android.content.Context

@SuppressLint("StaticFieldLeak")
object CoreModule {

    var appContext: Context? = null

    fun init(appContext: Context) {
        this.appContext = appContext
    }
}
